module/VuFind/src/VuFind/Bootstrapper.php

```diff
@@ -216,158 +217,6 @@ protected function initHeadTitle()
-            return;
-        }
-
-        $config = & $this->config;
-        $browserCallback = [$this, 'detectBrowserLanguage'];
-        $callback = function ($event) use ($config, $browserCallback) {
-            $validBrowserLanguage = call_user_func($browserCallback);
-
-            // Setup Translator
-            $request = $event->getRequest();
-            $sm = $event->getApplication()->getServiceManager();
-            if (($language = $request->getPost()->get('mylang', false))
-                || ($language = $request->getQuery()->get('lng', false))
-            ) {
-                $cookieManager = $sm->get('VuFind\Cookie\CookieManager');
-                $cookieManager->set('language', $language);
-            } elseif (!empty($request->getCookie()->language)) {
-                $language = $request->getCookie()->language;
-            } else {
-                $language = (false !== $validBrowserLanguage)
-                    ? $validBrowserLanguage : $config->Site->language;
-            }
-
-            // Make sure language code is valid, reset to default if bad:
-            if (!in_array($language, array_keys($config->Languages->toArray()))) {
-                $language = $config->Site->language;
-            }
-            try {
-                $translator = $sm->get('Zend\Mvc\I18n\Translator');
-                $translator->setLocale($language)
-                    ->addTranslationFile('ExtendedIni', null, 'default', $language);
-                foreach ($this->getTextDomains() as $domain) {
-                    // Set up text domains using the domain name as the filename;
-                    // this will help the ExtendedIni loader dynamically locate
-                    // the appropriate files.
-                    $translator->addTranslationFile(
-                        'ExtendedIni', $domain, $domain, $language
-                    );
-                }
-            } catch (\Zend\Mvc\I18n\Exception\BadMethodCallException $e) {
-                if (!extension_loaded('intl')) {
-                    throw new \Exception(
-                        'Translation broken due to missing PHP intl extension.'
-                        . ' Please disable translation or install the extension.'
-                    );
-                }
-            }
-            // Send key values to view:
-            $viewModel = $sm->get('ViewManager')->getViewModel();
-            $viewModel->setVariable('userLang', $language);
-            $viewModel->setVariable('allLangs', $config->Languages);
-            $rtlLangs = isset($config->LanguageSettings->rtl_langs)
-                ? array_map(
-                    'trim', explode(',', $config->LanguageSettings->rtl_langs)
-                ) : [];
-            $viewModel->setVariable('rtl', in_array($language, $rtlLangs));
-        };
-        $this->events->attach('dispatch.error', $callback, 10000);
-        $this->events->attach('dispatch', $callback, 10000);
-    }
-
```

Comment:
These methods moved into the newly introduced [`VuFind\I18n\Initializer`](https://github.com/vufind-org/vufind/blob/c5ec6f98ac86daef4200b50d3d65bbbd495371d4/module/VuFind/src/VuFind/I18n/Initializer.php#L177-L289).

Reply:
Much better -- good idea!